lib/rubocop/cop/rspec/described_class.rb

```diff
@@ -38,27 +52,42 @@ def on_block(node)
           describe, described_class, body = described_constant(node)
           return unless top_level_describe?(describe)
 
-          find_constant_usage(body, described_class) do |match|
-            add_offense(match, :expression, format(MSG, match.const_name))
+          @described_class = described_class
+
+          find_constant_usage(body) do |match|
+            add_offense(match, :expression, message(match.const_name))
           end
         end
 
         def autocorrect(node)
+          replacement = if style == :described_class
+                          DESCRIBED_CLASS
+                        else
+                          @described_class.const_name
+                        end
           lambda do |corrector|
-            corrector.replace(node.loc.expression, DESCRIBED_CLASS)
+            corrector.replace(node.loc.expression, replacement)
           end
         end
 
         private
 
-        def find_constant_usage(node, described_class, &block)
-          yield(node) if node.eql?(described_class)
+        def find_constant_usage(node, &block)
```

Comment:
I would rename this method now that it is looking for constant usage OR `described_class` usage.